File: python/1/1.py

def sumer(k):
    return k * (k + 1) / 2

def awesome(num):
    k = num - 1
    return 3*sumer(k/3) + 5*sumer(k/5) - 15*sumer(k/15)

print "Sum is: ", sum([i for i in range(1,1000) if i%5==0 or i%3==0])
print "Awe sum: ", awesome(1000)

File: c++/1/1.cpp
#include<iostream>
using namespace std;

int sum(int k)
{
  int sum = 0;
  for(int i=1; i<k; i++)
    {
      if ((i%5==0) || (i%3==0))
          sum +=i;
    }
  return sum;
}

//Better math solution

int summation(int k)
{
  return k *(k+1)/2;
}

int awesome(int a)
{
  int k = a - 1;
  return 3*summation(k/3) + 5*summation(k/5) - 15*summation(k/15);
}

int main()
{
  int number;
  std::cout<<"Enter a number: ";
  std::cin>>number;
  std::cout<<std::endl<<"Print sum trivial is: "<<sum(number);
  std::cout<<std::endl<<"Sum awesome is: "<<awesome(number)<<std::endl;
}
